crates/bevy_ecs/src/bundle.rs

```diff
@@ -0,0 +1,233 @@
+pub use bevy_ecs_macros::Bundle;
+
+use crate::{
+    component::{Component, ComponentFlags, ComponentId, Components, StorageType, TypeInfo},
+    entity::Entity,
+    storage::{SparseSetIndex, SparseSets, Table},
+};
+use bevy_ecs_macros::all_tuples;
+use std::{any::TypeId, collections::HashMap};
+
+/// A dynamically typed ordered collection of components
+///
+/// See [Bundle]
+pub trait DynamicBundle: Send + Sync + 'static {
+    /// Gets this [DynamicBundle]'s components type info, in the order of this bundle's Components
+    fn type_info(&self) -> Vec<TypeInfo>;
+
+    /// Calls `func` on each value, in the order of this bundle's Components. This will "mem::forget" the bundle
+    /// fields, so callers are responsible for dropping the fields if that is desirable.
+    fn put(self, func: impl FnMut(*mut u8));
```

Comment:
I think `put` and `get` should be swapped. In my mind `put` would put data into a bundle and `get` would get data out of a bundle. This does the exact opposite.

Reply:
Yeah that does make more sense doesn't it :)

Reply:
Maybe something a bit more descriptive like `dynamic_bundle.get_components()` and `Bundle::from_components()`?

Reply:
:+1: